gapic-common/lib/gapic/nonstandard_lro/operation.rb

```diff
@@ -0,0 +1,294 @@
+      #   long-running operation object. It is ignored if `operation_err_field` is given. Optional.
+      #
+      # @param operation_copy_fields [Map<String, String>] The map of the fields that need to be copied from the
+      #   long-running operation object that the polling method returns to the polling request.
+      #   The format is `name of the operation object field` -> `name of the request field` (`from` -> `to`)
+      #
+      # @param options [Gapic::CallOptions] call options for this operation
+      #
+      def initialize operation, client:, polling_method_name:, operation_status_field:,
+                     request_values: {}, operation_name_field: nil, operation_err_field: nil,
+                     operation_err_code_field: nil, operation_err_msg_field: nil, operation_copy_fields: {},
+                     options: {}
+        @client = client
+        @polling_method_name = polling_method_name
+        @operation_status_field = operation_status_field
+
+        @request_values = request_values || {}
+
+        @operation_name_field = operation_name_field
+        @operation_err_field = operation_err_field
+        @operation_err_code_field = operation_err_code_field
+        @operation_err_msg_field = operation_err_msg_field
+
+        @operation_copy_fields = operation_copy_fields || {}
+
+        @on_done_callbacks = []
+        @on_reload_callbacks = []
+        @options = options || {}
+
+        super operation
+      end
+
+      ##
+      # If the operation is done, returns the response. If the operation response is an error, the error will be
+      # returned. Otherwise returns nil.
+      #
+      # @return [Object, nil] The result of the operation or an error.
+      def results
+        return error if error?
+        return response if response?
+      end
+
+      ##
+      # Returns the name of the operation, if specified.
+      #
+      # @return [String, nil] The name of the operation.
+      #
+      def name
+        @operation.send @operation_name_field if @operation.respond_to? @operation_name_field
+      end
+
+      ##
+      # Checks if the operation is done. This does not send a new api call, but checks the result of the previous api
+      # call to see if done.
+      #
+      # @return [Boolean] Whether the operation is done.
+      #
+      def done?
+        return status if !status.nil? == status
+
+        status == :DONE
+      end
+
+      ##
+      # Checks if the operation is done and the result is not an error. If the operation is not finished then this will
+      # return false.
+      #
+      # @return [Boolean] Whether a response has been returned.
+      #
+      def response?
+        done? && !error?
+      end
+
+      ##
+      # If the operation is completed successfully, returns the underlying operation object, otherwise returns nil.
+      #
+      # @return [Object, nil] The response of the operation.
+      def response
+        @operation if response?
```

Comment:
Use the `operation` method rather than `@operation`

Reply:
:+1: